dependency updates and target SDK 29

nickbw2003 opened the pull request.

nickbw2003 committed:
update dependencies

nickbw2003 committed:
define dependency versions in separate gradle file

nickbw2003 committed:
fix: deprecation warning of PreferenceManager

nickbw2003 committed:
fix method override

nickbw2003 committed:
Android Q doesn't provide PermissionGroupInfos anymore, therefore we will fallback to the PermissionInfo

nickbw2003 committed:
use string plurals for departure list item's departure time

nickbw2003 committed:
switching to another network has to change the default location

nickbw2003 committed:
restrict backups

nickbw2003 committed:
remove unused resource

nickbw2003 committed:
fix missing contentDescription warning

nickbw2003 committed:
ignore overdraw warning

nickbw2003 committed:
fix missing contentDescription warning

nickbw2003 committed:
ignore long vector path warning

akoufa commented on app/src/main/kotlin/de/nickbw2003/stopinfo/map/data/LocationService.kt:
```diff
@@ -25,7 +25,7 @@ class LocationService(context: Context, networkRepository: NetworkRepository) {
             if (it != null) {
-                currentLocation.defaultLocation = Location(it.lat, it.lng)
+                currentLocation.defaultLocation = Location(it.lat, it.lng, true)
             }
```
I would use named parameters here as it is easy to mismatch lat and lng

akoufa reviewed (commented):
I left a comment about one line in the LocationService.kt file

nickbw2003 committed:
Update buildtools version

nickbw2003 committed:
Update androidx dependencies

nickbw2003 committed:
Remove unnecessary definition of coroutine scope in ViewModel

As we updated to androidx.lifecycle 2.1 we can make use of the viewModelScope extension property instead of defining our own coroutine scope

nickbw2003 committed:
Use named params for new instance creation of Location

To avoid mismatching of lat and lng values
